synapse/tests/common.py

```diff
@@ -25,6 +28,14 @@
 
 class TooFewEvents(Exception): pass
 
+# Py2/3 SSL Exception Compat
+if s_compat.version >= (3, 0, 0):
```

Comment:
From the discussion with @vertexmc - this was a case where the error was specific to running the tests across python versions; and the errors would be seen client side (ie. the thing talking to webapp), not inside of synapse itself.